MetaMask/metamask-mobile
# feat: upgrade assets controllers to 42 with multichain token rates

## **Description**

This PR upgrades asset controllers to 42.  This version contains `resetState` functions in controllers, so it no longer needs to be patched.  It also allows us to poll erc20 token rates across chains.

## **Related issues**



## **Manual testing steps**

With PORTFOLIO_VIEW=false and PORTFOLIO_VIEW=true:

1. Import some erc20 tokens
2. Verify erc20 token prices are correct
3. Switch chains
4. Verify erc20 token prices are correct
5. In settings, change the fiat currency
6. Verify erc20 token prices are correct

## **Screenshots/Recordings**



### **Before**



### **After**

No visual changes.  erc20 prices should be correct when adding tokens, switching networks, and switching currencies.

https://github.com/user-attachments/assets/29a733ae-630f-4d1f-952c-d5ac54bef14a



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
